File: backend/models/microbe.js
module.exports = (sequelize, DataTypes) => {
  const Microbe = sequelize.define('Microbe', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: false,
      allowNull: false,
    },

    // English
    common_name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    type: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    lecture_text: {
      type: DataTypes.TEXT,
      allowNull: false,
    },
    feedback_correct: {
      type: DataTypes.TEXT,
      allowNull: false,
    },
    feedback_incorrect: {
      type: DataTypes.TEXT,
      allowNull: false,
    },

    // Swedish
    common_name_sv: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    type_sv: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    lecture_text_sv: {
      type: DataTypes.TEXT,
      allowNull: false,
    },
    feedback_correct_sv: {
      type: DataTypes.TEXT,
      allowNull: false,
    },
    feedback_incorrect_sv: {
      type: DataTypes.TEXT,
      allowNull: false,
    },

    // Language-independent
    scientific_name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    bsl_level: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
  }, {
    tableName: 'microbes',
    timestamps: false,
  });

  return Microbe;
};

File: backend-python/app/models/microbe.py
from sqlalchemy import Column, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship

from app.database import Base


class Microbe(Base):
    __tablename__ = "microbes"

    id = Column(Integer, primary_key=True)

    common_name = Column(String, nullable=False)

    scientific_name = Column(String, nullable=False)

    type = Column(String, nullable=False)

    bsl_level = Column(
        Integer,
        ForeignKey("bsl_classes.class_number")
    )

    lecture_text = Column(Text, nullable=False)

    feedback_correct = Column(Text, nullable=False)

    feedback_incorrect = Column(Text, nullable=False)

    bsl_class = relationship(
        "BSLClass",
        back_populates="microbes"
    )